storage2/src/state.rs

```diff
@@ -29,14 +34,22 @@ impl State {
     pub fn begin_transaction(&mut self) -> StateTransaction {
         StateTransaction::new(self)
     }
+
+    // Apply the unwritten changes of a transaction to this state fork.
+    pub fn apply_transaction(&mut self, transaction: StateTransaction) {
+        for (key, value) in transaction.unwritten_changes.iter() {
+            self.unwritten_changes.insert(key.clone(), value.clone());
+        }
```

Comment:
This could be rewritten as:

```rust
self.unwritten_changes.extend(transaction.unwritten_changes)
```

This has the advantage of avoiding cloning, which you don't need to do because `StateTransaction` is taken by value.